### PR TITLE
tendermint unconditional peers

### PR DESCRIPTION
**Pull request by ptrus**
Expose new p2p config options  which were added in tendermint 0.33

### CODE REVIEW

File: .changelog/2855.feature.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+go/consensus/tendermint: Expose new config options added in Tendermint 0.33
+
+Tendermint 0.33 added the concept of unconditional P2P peers. Support for
+setting the unconditional peers via `tendermint.p2p.unconditional_peer_ids`
+configuration flag is added. On sentry node, upstream nodes will automatically
+be set as uncodnitional peers.
```
Comment (kostko) [inline .changelog/2855.feature.md:6]:
```suggestion
be set as unconditional peers.
```

File: .changelog/2855.feature.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+Tendermint 0.33 added support for setting maximum re-dial period when
```
Comment (kostko) [inline .changelog/2855.feature.md:7]: New line above this line.

File: go/consensus/tendermint/tendermint.go [hunk 3]
```diff
@@ -1377,8 +1386,10 @@ func init() {
 	Flags.Uint64(cfgABCIPruneNumKept, 3600, "ABCI state versions kept (when applicable)")
 	Flags.StringSlice(CfgSentryUpstreamAddress, []string{}, "Tendermint nodes for which we act as sentry of the form ID@ip:port")
 	Flags.StringSlice(CfgP2PPersistentPeer, []string{}, "Tendermint persistent peer(s) of the form ID@ip:port")
+	Flags.StringSlice(CfgP2PUnconditionalPeerIDs, []string{}, "Tendermint unconditional peer IDs")
 	Flags.Bool(CfgP2PDisablePeerExchange, false, "Disable Tendermint's peer-exchange reactor")
 	Flags.Bool(CfgP2PSeedMode, false, "run the tendermint node in seed mode")
+	Flags.Duration(CfgP2PPersistenPeersMaxDialPeriod, 0*time.Second, "Tendemint max timeout when redialing a persistent peer (default unlimited).")
```
Comment (kostko) [inline go/consensus/tendermint/tendermint.go:1392]:
```suggestion
	Flags.Duration(CfgP2PPersistenPeersMaxDialPeriod, 0*time.Second, "Tendermint max timeout when redialing a persistent peer (default: unlimited)")
```

File: go/consensus/tendermint/tendermint.go [hunk 4]
```diff
@@ -1377,8 +1386,10 @@ func init() {
 	Flags.Uint64(cfgABCIPruneNumKept, 3600, "ABCI state versions kept (when applicable)")
 	Flags.StringSlice(CfgSentryUpstreamAddress, []string{}, "Tendermint nodes for which we act as sentry of the form ID@ip:port")
 	Flags.StringSlice(CfgP2PPersistentPeer, []string{}, "Tendermint persistent peer(s) of the form ID@ip:port")
+	Flags.StringSlice(CfgP2PUnconditionalPeerIDs, []string{}, "Tendermint unconditional peer IDs")
 	Flags.Bool(CfgP2PDisablePeerExchange, false, "Disable Tendermint's peer-exchange reactor")
 	Flags.Bool(CfgP2PSeedMode, false, "run the tendermint node in seed mode")
+	Flags.Duration(CfgP2PPersistenPeersMaxDialPeriod, 0*time.Second, "Tendermint max timeout when redialing a persistent peer (default: unlimited).")
```
Comment (kostko) [inline go/consensus/tendermint/tendermint.go:1392]: You overlooked removing the dot at the end :-)

Comment (ptrus) [reply go/consensus/tendermint/tendermint.go]: ah.. :)

**Pull request merged**